csharp/src/Apache.Arrow/Arrays/ArrayData.cs

```diff
@@ -30,31 +30,34 @@ public sealed class ArrayData : IDisposable
         public readonly int Offset;
         public readonly ArrowBuffer[] Buffers;
         public readonly ArrayData[] Children;
+        public readonly ArrayData Dictionary; //Only used for dictionary type
 
         public ArrayData(
             IArrowType dataType,
             int length, int nullCount = 0, int offset = 0,
-            IEnumerable<ArrowBuffer> buffers = null, IEnumerable<ArrayData> children = null)
+            IEnumerable<ArrowBuffer> buffers = null, IEnumerable<ArrayData> children = null, ArrayData dictionary = null)
```

Comment:
These new parameters cause a binary breaking change. If someone was built against the old version, and called `new ArrayData(..., ...);`, when that already built assembly runs on the new version, they will get `MissingMemberException` because the old version of the ctor isn't there.

These will need to be new ctor overloads in order to avoid the breaking change.

Reply:
Fixed it.